### DISCUSSION TITLE
Improvements to capsule

### DISCUSSION THREAD

**Discussion by XuJiandong** [nervosnetwork/capsule#46] [Ideas]
1. Docker's [network mode](https://docs.docker.com/network/) can be switched to "host" by flags "--host". 

It will make configuring proxy easier. Also update the wiki: [Proxy-Configuration](https://github.com/nervosnetwork/capsule/wiki/Proxy-Configuration) when host mode is on.

2. Pass arguments to cargo build

All arguments followed by "--" can be passed to cargo inside docker while using `capsule build`. 

3. Make "capsule test" deprecated
Print warning when users use "capsule test": 
```
Capsule test subcommand is deprecated. Use `cargo test -p <...>` instead.
```

4. Pass environment variables in command line `cargo build`

``` 
--env-file list                  Read in a file of environment variables
```

It will be used by docker directly.

**Comment by TheWaWaR** [comment]
Implemented in https://github.com/nervosnetwork/capsule/pull/57
